src/material/form-field/form-field.ts

```diff
@@ -426,10 +428,22 @@ export class MatFormField
     this._stateChanges?.unsubscribe();
     this._stateChanges = control.stateChanges.subscribe(() => {
       this._updateFocusState();
-      this._syncDescribedByIds();
       this._changeDetectorRef.markForCheck();
     });
 
+    // Updating the `aria-describedby` touches the DOM. Only do it if it actually needs to change.
+    this._describedByChanges?.unsubscribe();
+    this._describedByChanges = control.stateChanges
```

Comment:
I tried basing this on top of signals, but the timing was off because I had to run `_syncDescribedByIds` in an effect.

Reply:
Ideally we'll do a bigger refactor at some point and replace `syncDescribedByIds` with a signal, `describedByIds`

Reply:
Yep, I'd argue pretty much all the properties in the `FormFieldControl` should be signals.